### PR TITLE
Address book submodule 

### PR DESCRIPTION
**Pull request by laics1984**
- Applied models and refactoring in submodule address book 
- Reposition of import link/button

### CODE REVIEW

File: src/modules/services/submodule/addressbook/views/ViewServicesAddressBookAddContacts.vue [hunk 1]
```diff
@@ -64,27 +69,43 @@ export default {
...
+      const contactAddIndex = (wallet.contacts!=undefined)?wallet.contacts.findIndex((contact) => contact.address == formattedAddress):(-1);
+      const contactNameIndex =(wallet.contacts!=undefined)?wallet.contacts.findIndex((contact) => contact.name.toLowerCase() == contactName.value.toLowerCase()):(-1);
+
+      if(contactAddIndex >= 0 || accountAddIndex >= 0){
+        err.value = 'Address is already exists in account or address book.';
+      }else if( contactNameIndex >= 0 || accountNameIndex >= 0 ){
+        err.value = 'Name is already exists in account or address book.';
+      }else{
+        walletState.currentLoggedInWallet.addAddressBook(addressBook);
+        walletState.wallets.savetoLocalStorage();
```
Comment (sleepyOwl14) [inline src/modules/services/submodule/addressbook/views/ViewServicesAddressBookAddContacts.vue:96]: please don't use savetoLocalStorage, please use 
```
walletState.wallets.saveMyWalletOnlytoLocalStorage(walletState.currentLoggedInWallet)
```

File: src/modules/services/submodule/addressbook/components/EditContactModal.vue [hunk 2]
```diff
@@ -76,28 +79,27 @@ export default{
...
-      }else if(status === -1){
-        err.value = 'Edit operation failed';
-      }else{
-        err.value = status;
-      }
+      // @param index, AddressBook
+      const contactIndex = walletState.currentLoggedInWallet.contacts.findIndex((contact) => contact.address == p.data.address);
+      // const walletInstance = new Wallet();
+      walletState.currentLoggedInWallet.updateAddressBook(contactIndex, { name: contactName.value, address: address.value });
+      walletState.wallets.savetoLocalStorage();
```
Comment (sleepyOwl14) [inline src/modules/services/submodule/addressbook/components/EditContactModal.vue:99]: please don't use savetoLocalStorage, please use 
```
walletState.wallets.saveMyWalletOnlytoLocalStorage(walletState.currentLoggedInWallet)
```

File: src/modules/services/submodule/addressbook/components/ConfirmDeleteContactModal.vue [hunk 3]
```diff
@@ -42,24 +42,26 @@
...
     const toggleModal = ref(false);
     const deleteContact = (contact) => {
-      var deleteStatus = appStore.deleteContact(contact);
-      if(deleteStatus){
-        toggleModal.value = !toggleModal.value;
-        emitter.emit('REFRESH_CONTACT_LIST', true);
-      }
+      const contactIndex = walletState.currentLoggedInWallet.contacts.findIndex((contact) => contact.address == p.data.address);
+      walletState.currentLoggedInWallet.removeAddressBook(contactIndex);
+      walletState.wallets.savetoLocalStorage();
```
Comment (sleepyOwl14) [inline src/modules/services/submodule/addressbook/components/ConfirmDeleteContactModal.vue:62]: please don't use savetoLocalStorage, please use 
```
walletState.wallets.saveMyWalletOnlytoLocalStorage(walletState.currentLoggedInWallet)
```

File: src/modules/services/submodule/addressbook/views/ViewServicesAddressBookAddContacts.vue (hunk 1 above)
Comment (laics1984) [reply src/modules/services/submodule/addressbook/views/ViewServicesAddressBookAddContacts.vue:96]: Okay noted with thanks

**Review by sleepyOwl14 — CHANGES_REQUESTED** [review]
please don't use savetoLocalStorage, please use 
```
walletState.wallets.saveMyWalletOnlytoLocalStorage(walletState.currentLoggedInWallet)
```

File: src/modules/services/submodule/addressbook/components/EditContactModal.vue (hunk 2 above)
Comment (laics1984) [reply src/modules/services/submodule/addressbook/components/EditContactModal.vue:99]: Okay noted with thanks

File: src/modules/services/submodule/addressbook/components/ConfirmDeleteContactModal.vue (hunk 3 above)
Comment (laics1984) [reply src/modules/services/submodule/addressbook/components/ConfirmDeleteContactModal.vue:62]: Okay noted, changing in a moment

**Comment by laics1984** [comment]
> please don't use savetoLocalStorage, please use
> 
> ```
> walletState.wallets.saveMyWalletOnlytoLocalStorage(walletState.currentLoggedInWallet)
> ```

SavetoLocalStorage method updated

File: src/modules/services/submodule/addressbook/components/EditContactModal.vue [hunk 4]
```diff
@@ -33,11 +33,16 @@
   </div>
 </template>
 
-<script>
-import { computed, getCurrentInstance, inject, ref, watch } from 'vue';
+<script lang="ts">
```
Comment (sleepyOwl14) [inline src/modules/services/submodule/addressbook/components/EditContactModal.vue:36]: can remove this lang="ts" ?

Comment (laics1984) [reply src/modules/services/submodule/addressbook/components/EditContactModal.vue]: ts removed

**Review by sleepyOwl14 — APPROVED** [review]
good to go

**Pull request merged**